pudb/debugger.py

```diff
@@ -342,6 +342,22 @@ def set_frame_index(self, index):
 
         self.ui.stack_list._w.set_focus(self.ui.translate_ui_stack_index(index))
 
+    def open_file_to_edit(self, index):
+        self.curframe, line_number = self.stack[index]
+        filename = self.curframe.f_code.co_filename
+
+        editor = os.environ.get("EDITOR", "nano")
+        editor = editor.lower()
+
+        editor_command = f"{editor}"
+        if editor in {"vi", "vim", "nvim", "emacs"}:
+            editor_command = f"{editor} +{line_number}"
+        elif editor == "nano":
+            editor_command = f"{editor} -l {line_number}"
+
+        command_to_run = f"{editor_command} {filename}"
+        os.system(command=command_to_run)
```

Comment:
What completion semantics do we expect here? Should the source be reloaded when we come back from the editor? (This involves two possible parts: `linecache` and Python module reload) (If we do nothing, it'll be stale, which will be confusing.)

Reply:
Yeah, I'm unsure about this too. Doing a `reload` is probably a bad idea, as it is wrong even more subtle ways (if you really want to do that, you can do it yourself manually in the console). Maybe after editing it should popup a message asking if you want to stop the debugger so you can restart the interpreter?

Reply:
I think not using `reload` is the better option. I will display a message, so the user is aware that the file changed, and they can stop the execution and restart if they want

Reply:
DONE

Reply:
You could *offer* a reload in the message. Do you think that's reasonable?

Reply:
(This conversation is outdated)